doc/source/whatsnew/v1.0.0.rst

```diff
@@ -292,6 +293,32 @@ New repr for :class:`~pandas.arrays.IntervalArray`
 
    pd.arrays.IntervalArray.from_tuples([(0, 1), (2, 3)])
 
+- :meth:`DataFrame.merge` now preserves right frame's row order when executing a right merge (:issue:`27453`)
```

Comment:
make this a sub-section (need a line under the title), and put a shorter title, move what you have to the first sentence.